Framework/DataHandling/src/LoadEmptyInstrument.cpp

```diff
@@ -30,6 +35,31 @@ using namespace Geometry;
 using namespace DataObjects;
 using namespace HistogramData;
 
+namespace {
+const std::vector<std::string> validFilenameExtensions{".xml", ".hdf5", ".nxs", ".nxs.h5"};
+enum class FilenameExtensionEnum { XML, HDF5, NXS, NXS_H5, enum_count };
+typedef EnumeratedString<FilenameExtensionEnum, &validFilenameExtensions, &compareStringsCaseInsensitive>
+    FilenameExtension;
+// This method attempts to retrieve an enumerated file name extension. If it fails, an std::runtime _error will be
+// thrown.
+FilenameExtension getValidFilenameExtension(const std::string &filename) {
+  std::string ext{std::filesystem::path(filename).extension().string()};
+  // Note, a file name might have a double extension such as ".nxs.h5", which is valid.
+  // It may also have a double extension such as .lite.nxs, which is invalid.
+  // In the latter case a single extension, .nxs, which is valid, should be retrieved.
+  std::string stem{std::filesystem::path(filename).stem().string()};
+  std::string pre_ext{std::filesystem::path(stem).extension().string()};
+  if (!pre_ext.empty()) {
+    std::string double_ext{pre_ext + ext};
+    try {
+      return FilenameExtension(double_ext);
+    } catch (std::runtime_error &) {
+    }
+  }
+  return FilenameExtension(ext);
+}
```

Comment:
This is a standalone function.  Can you write some tests calling this with several filename strings such as "fakefile.nxs.h5" and "fakefile.snap.lite.hxs" etc, and verify that the correct extensions are returned?  It should not involved invoking the actual algorithm.